## Import Libraries

In [2]:
import pandas as pd
import numpy as np
import os

## Import Data

In [4]:
path = r"C:\Users\corri\Documents\New Career\Achievment 4\Instacart Basket Analysis\Data"

In [5]:
df_ords = pd.read_csv(os.path.join(path,"Original Data", "4.3_orders_products", "orders.csv"), index_col = False)

In [14]:
df_prds = pd.read_csv(os.path.join(path, "Prepared Data", "products_cleaned.csv"), index_col = False)

## Clean Data

In [15]:
df_prds.describe()

Unnamed: 0    product_id      aisle_id  department_id        prices
count  49672.000000  49672.000000  49672.000000   49672.000000  49672.000000
mean   24852.005053  24850.349775     67.762442      11.728942      9.993282
std    14342.265579  14340.705287     38.315784       5.850779    453.615536
min        0.000000      1.000000      1.000000       1.000000      1.000000
25%    12432.750000  12432.750000     35.000000       7.000000      4.100000
50%    24851.500000  24850.500000     69.000000      13.000000      7.100000
75%    37272.250000  37268.250000    100.000000      17.000000     11.100000
max    49692.000000  49688.000000    134.000000      21.000000  99999.000000

The minimum prices are 1 which seems unusual. Nothing is cheaper than 1.00 and the cheapest item is exactly 1.00.
The maximum unnamed and product_id are higher than the count which shouldn't be the case.  The maximum prices are $99,999.00
which seems much too high for a grocery store.

In [19]:
# Check df_ords for mixed data - there does not appear to be any
for col in df_ords.columns.tolist():
    weird = (df_ords[[col]].applymap(type) != df_ords[[col]].iloc[0].apply(type)).any (axis = 1) 
    if len (df_ords[weird])>0:
        print(col)

In [21]:
# Check df_ords for missing values
df_ords.isnull().sum()

order_id                       0
user_id                        0
eval_set                       0
order_number                   0
order_dow                      0
order_hour_of_day              0
days_since_prior_order    206209
dtype: int64

days_since_prior_order has 206,209 missing values.  This could be because this was the first order for that customer (so they don't have any prior orders).  I have chosen to ignore these missing values.  This information could be useful for finding first time customers.

In [25]:
# Check orders dataframe for duplicates
df_dups = df_ords[df_ords.duplicated()]

In [26]:
df_dups

Empty DataFrame
Columns: [order_id, user_id, eval_set, order_number, order_dow, order_hour_of_day, days_since_prior_order]
Index: []

There are no duplicates in df_ords.

In [27]:
df_ords.to_csv(os.path.join(path, "Prepared Data", "orders_cleaned.csv"))